packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -190,6 +190,11 @@ function SelectControl< ItemType = DefaultItemType >( {
 						};
 					}
 					break;
+				case selectControlStateChangeTypes.ControlledPropUpdatedSelectedItem:
+					if ( changes?.selectedItem ) {
+						setInputValue( getItemLabel( changes.selectedItem ) );
```

Comment:
This is a `stateReducer` anything within this function should stay pure, so we shouldn't make a call to `setInputValue` that updates the state.
You should be able to do this instead:
```
newChanges = {
							...changes,
							inputValue: getItemLabel( changes.selectedItem ),
						};
```

Also I was curious why you needed this? Is this to make sure the label updates if you update the selected item from outside of the SelectControl component?
If that is the case, when does this happen?
As it should set the input value for a pre-selected item already correctly.

Reply:
Definitely true, although this "fix" doesn't work when done via that method for some reason. Here's the issue:

![Peek 2022-10-31 16-04](https://user-images.githubusercontent.com/444632/199126137-f5d322b7-f404-4a0b-9665-76ef0af50ddc.gif)

Basically the `inputValue` for the field remains the value of the last attribute when a new one is selected.

In short the `inputValue` provided to the `onInputValueChange` is out-of-date which causes the original issue, and isn't fired again after the above reducer, which prevents your fix from working. Setting the input value directly from the reducer worked, although it's a misplaced side-effect. 

This "feels" like a `downshift` bug, although I'm not very familiar with the package. If you have any insights on another solution let me know. I'll take a closer look at this now, regardless.

Reply:
Thanks @louwie17 , looks like this is not an issue with the design changes.